feat: Add a sample page for referral data type

mridulgrc opened the pull request:
- Added a sample page "/things/referral" for referral data type

Relates to: CHBASE-8006

mridulgrc committed:
feat: Add example for referral data-type

- Added a sample page for referral data type

Relates to: CHBASE-8006

rajeevs1992 commented on examples/ui-jaxb/src/main/java/com/chbase/applications/referral/ReferralPage.java:
```diff
@@ -0,0 +1,166 @@
+/*
+testme * Copyright 2011 Microsoft Corp.
```
copyright should be GRh

mridulgrc committed:
fix: Update Copyright text.

Added GRH's Copyright text "Copyright (c) 2023 Get Real Health.  All rights reserved"
